go-controller/pkg/metrics/master.go

```diff
@@ -300,3 +307,48 @@ func RecordSubnetCount(v4SubnetCount, v6SubnetCount float64) {
 	metricV4HostSubnetCount.Set(v4SubnetCount)
 	metricV6HostSubnetCount.Set(v6SubnetCount)
 }
+
+// chassisE2EScraper implements the Collector interface so we can
+// generate chassis metrics on-the-fly per scrape
+type chassisE2EScraper struct {
+	sbClient goovn.Client
+}
+
+func (c *chassisE2EScraper) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(c, ch)
+}
+
+// Collect scrapes the Chassis and Chassis_Priv tables to determine the
+func (c *chassisE2EScraper) Collect(ch chan<- prometheus.Metric) {
+	// the nb_cfg value is in Chassis_Private, but the
+	// hostname is in Chassis. So we need to list both and join
+	allChassis, err := c.sbClient.ChassisList()
+	if err != nil {
+		klog.Errorf("Failed to list sbdb Chassis table: %v", err)
+		return
+	}
+
+	// map of chassis name to hostname
+	hostnames := make(map[string]string, len(allChassis))
+	for _, chassis := range allChassis {
+		hostnames[chassis.Name] = chassis.Hostname
+	}
+
+	allChassisPriv, err := c.sbClient.ChassisPrivateList()
+	if err != nil {
+		klog.Errorf("Failed to list sbdb Chassis_Private table: %v", err)
+	}
+	klog.V(5).Infof("Got %d chassis rows", len(allChassis))
```

Comment:
Perhaps, we should remove L341 as well even though it is set to V(5) or we should increase the logging to V(7) or something..

Reply:
I deleted it. There's no need to log it.